### DISCUSSION TITLE
Higher Level Level 2 ImproPhe

### DISCUSSION THREAD

**Discussion by kribeetz** [davidfrantz/force#255] [Q&A]
Hello everyone, 

I'm trying to use the force higher level Level 2 ImproPhe especially for resampling my Landsat data to 10m resolution and to spectral harmonize both. I am working with Landsat 8,9 and Sentinel 2A/B data with about 16 images from mid July to mid August of 2022. There are already in Level 2 ARD.

Therefore I created a copy of some data (6 days, from 11/07 to 28/07 for both satellites) into a lower directory folder with a .prj file and a higher directory where the parameter file is laying. Everytime I try to use the command force-higher-level l2imp.prm the processing is done within a second and doesn't process any image. It only creates this CITEME.txt and another .prj file in the higher level directory. You can see the screenshot of it here: 

<img width="466" alt="folder_structure" src="https://user-images.githubusercontent.com/118894630/203724269-2087e990-0333-45b9-afb1-93697898430d.png">

The .prj file is also in the X0033_Y0025 folder. When I put it into the lower directory (one folder above) it will give me the error message: _Reading datacube definition failed. Copying datacube definition failed._

And a screenshot of the command line output here: 
<img width="640" alt="force_higher_level" src="https://user-images.githubusercontent.com/118894630/203722500-209efc41-9929-4b95-87a1-fe928c86550e.png">

I specified all the following option in the parameter file:
- Input and Output Directory
- No Masking
- Output Options: GTiff
- Adjusted the parallel processing
- Processing Extent: Tiles are only covering X0033_Y0025
- Set Sensor Allow_list to Sen2A, Sen2B, LND08, LND09
- Selected some QAI for screening
- Adjusted the timeframe to DOY = 91 273 (so this should contain all the days I'm covering) and DATE_RANGE = 2022-01-01 2022-12-31
- Sensor Low Resolution: LND08 LND09 

With the ImproPhe Parameters I'm not to sure what to put exactly. Maybe you can take a look into the parameter file and give me a suggestion on what to change. (I exported the .prm file to .txt so it could be uploaded)

Thank you!
Kristina
[l2imp.txt](https://github.com/davidfrantz/force/files/10081810/l2imp.txt)

**Comment by davidfrantz** [comment]
Hi Kristina,

I have to admit that I didn't look into ImproPhe for a while, but please try following things:

``DIR_LOWER`` needs to be the directory of the whole datacube, not one tile:

DIR_LOWER = /home/user/X0033_Y0025 --> DIR_LOWER = /home/user/lower_dir

lower_dir should contain datacube-definition.prj and X0033_Y0025. In X0033_Y0025, there shouldn't be a datacube-definition.prj file.

``DIR_HIGHER`` should be the same folder as ``DIR_LOWER``

``SENSORS`` should only contain the high-res Sentinel-2 satellites, i.e. sen2a sen2b (small letters), and please disable ``SPECTRAL_ADJUST``

The ``SEASONAL_WINDOW`` does not make too much sense, as you are only working with a seasonal subset anyway. The idea behind ImproPhe is to use all base images within a year, and generate seasonal slices. You could try to use July only, but in this case, only one window will suffice. For the other ImproPhe parameters, I guess the default values are mostly OK, but you might need to adapt ``KERNEL_SIZE`` to be a multiple of 30m, e.g. 3000

I hope this helps,
David
